module/Finna/src/Finna/Search/DateRangeFilterTrait.php

```diff
@@ -0,0 +1,197 @@
+        [$field, ] = $this->parseFilter($filter);
+
+        if (
+            $field == $this->getDateRangeSearchField()
+            || $field == $this->spatialDateRangeFieldVF1
+        ) {
+            // Date range filters are processed
+            // separately (see initSpatialDateRangeFilter)
+            return;
+        }
+        parent::addFilter($filter);
+    }
+
+    /**
+     * Return current date range filter.
+     *
+     * @return mixed false|array Filter
+     */
+    public function getDateRangeFilter(): mixed
+    {
+        $filterList = $this->getFilterList();
+        foreach ($filterList as $filters) {
+            foreach ($filters as $filter) {
+                if ($this->isDateRangeFilter($filter['field'])) {
+                    return $filter;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Return the current filters as an array of strings ['field:filter']
+     *
+     * @return array $filterQuery
+     */
+    public function getFilterSettings(): array
+    {
+        $result = parent::getFilterSettings();
+
+        // Special processing for date range filters
+        $dateRangeField = $this->getDateRangeSearchField();
+        if ($dateRangeField) {
+            foreach ($result as &$filter) {
+                $dateRange = strncmp(
+                    $filter,
+                    "$dateRangeField:",
+                    strlen($dateRangeField) + 1
+                ) == 0;
+                if ($dateRange) {
+                    [, $value] = $this->parseFilter($filter);
+                    [$op, $range] = explode('|', $value);
+                    $op = $op == 'within' ? 'Within' : 'Intersects';
+                    $filter = "{!field f=$dateRangeField op=$op}$range";
+                }
+            }
+        }
+        return $result;
+    }
+
+    /**
+     * Initialize date range filter (search_daterange_mv)
+     *
+     * @param \Laminas\Stdlib\Parameters $request Parameter object representing user
+     * request.
+     *
+     * @return void
+     */
+    public function initSpatialDateRangeFilter($request): void
+    {
+        $dateRangeField = $this->getDateRangeSearchField();
+        if (!$dateRangeField) {
+            return;
+        }
+        $type = $request->get("{$dateRangeField}_type");
+        if (!$type) {
+            // VuFind 1
+            $type = $request->get($this->spatialDateRangeFieldTypeVF1);
+        }
+        if (!$type) {
+            $type = $this->spatialDateRangeDefaultType;
+        }
+
+        $from = $to = null;
+        $found = false;
+        // Date range filter
+        if (($reqFilters = $request->get('filter')) && is_array($reqFilters)) {
+            foreach ($reqFilters as $f) {
+                [$field, ] = $this->parseFilter($f);
+                if (
+                    $field == $dateRangeField
+                    || $field == $this->spatialDateRangeFieldVF1
```

Comment:
sais yhen sisäkkäisen iffin pois jos laittais viel (field == $daterangefield || field == this->spatialDateRange..) && $range = $this->parseDateRangeFilter($f) samaan

Reply:
tai jopa in_array($field, [$dateRangeField, $this->spatialDateRangeFieldVF1]) && $range = $this->parse...